shims/0.23/src/main/java/org/apache/hadoop/hive/shims/Hadoop23Shims.java

```diff
@@ -1211,6 +1212,35 @@ public boolean runDistCpWithSnapshots(String oldSnapshot, String newSnapshot, Li
     return false;
   }
 
+  protected int runDistCpInternal(DistCp distcp, List<String> params) {
+    ensureMapReduceQueue(distcp.getConf());
+    return distcp.run(params.toArray(new String[0]));
+  }
+
+  /**
+   * This method ensures if there is an explicit tez.queue.name set, the hadoop shim will submit jobs
+   * to the same yarn queue. This solves a security issue where e.g settings have the following values:
+   * tez.queue.name=sample
+   * hive.server2.tez.queue.access.check=true
+   * In this case, when a query submits Tez DAGs, the tez client layer checks whether the end user has access to
+   * the yarn queue 'sample' via YarnQueueHelper, but this is not respected in case of MR jobs that run
+   * even if the query execution engine is Tez. E.g. an EXPORT TABLE can submit DistCp MR jobs at some stages when
+   * certain criteria are met. We tend to restrict the setting of mapreduce.job.queuename in order to bypass this
+   * security flaw, and even the default queue is unexpected if we explicitly set tez.queue.name.
+   * Under the hood the desired behavior is to have DistCp jobs in the same yarn queue as other parts
+   * of the query. Most of the time, the user isn't aware that a query involves DistCp jobs, hence isn't aware
+   * of these details.
+   */
+  protected void ensureMapReduceQueue(Configuration conf) {
+    String queueName = conf.get(TezConfiguration.TEZ_QUEUE_NAME);
+    LOG.debug("Checking tez.queue.name {}", queueName);
+    if (queueName != null && queueName.length() > 0) {
+      LOG.info("Setting mapreduce.job.queuename (current: '{}') to become tez.queue.name: '{}'",
+          conf.get(MRJobConfig.QUEUE_NAME), queueName);
+      conf.set(MRJobConfig.QUEUE_NAME, queueName);
+    }
```

Comment:
I have doubts here, Distcp itself is a MapReduce Job, it will launch a MR job only, so why it should not use MR queue?
Second, Hive on MR might not be widely used but still exists and we haven't mentioned anywhere it won't work upstream.
So, if someone isn't using Tez, will this create problems?

Reply:
Say during replication, External Table are copied they are simple DistCp jobs, they should use that only?
Second, is this an incompatible change as well?

Reply:
```
So, if someone isn't using Tez, will this create problems?
```
not really, this doesn't affect the original MR query stages, only DistCp, also we don't support MR anymore

honestly, I'm not sure about external table replication, so you're right, maybe it's not correct to introduce this behavior globally on hadoop shim distcp level, instead apply only to the target codepath, which is the EXPORT TABLE, would it make more sense?

Reply:
> which is the EXPORT TABLE, would it make more sense?

yep, that makes sense to me. If possible better to check if the query engine is tez and may be then do this